### PR TITLE
[PREGEL] get rid of promises

### PR DESCRIPTION
**Pull request by jvolmer**
- Moves GraphStore into Worker directory
- Gets rid of promises: instead, uses futures::makeFutureWith which makes a real future out of a function.

### CODE REVIEW

File: arangod/Pregel/Worker/Worker.cpp [hunk 1]
```diff
@@ -787,7 +787,9 @@ auto Worker<V, E, M>::loadGraph(LoadGraph const& graph)
 
   LOG_PREGEL("52070", WARN) << fmt::format(
       "Worker for execution number {} is loading", _config.executionNumber());
-  return std::move(_graphStore->loadShards(&_config, _makeStatusCallback()))
+  return futures::makeFutureWith([&]() {
```
Comment (MBkkt) [inline arangod/Pregel/Worker/Worker.cpp:790]: Why not `futures::makeFuture(_graphStore->loadShards(&_config, _makeStatusCallback()))`?

**Review by MBkkt — APPROVED** [review]

**Review by MBkkt — CHANGES_REQUESTED** [review]
Why do you create std::function and then call it instead of create future directly

File: arangod/Pregel/Worker/Worker.cpp [hunk 2]
```diff
@@ -363,7 +367,8 @@ auto Worker<V, E, M>::_processVerticesInThreads() -> VerticesProcessedFuture {
     TRI_ASSERT(endI <= numSegments);
 
     auto vertices = _graphStore->vertexIterator(startI, endI);
-    processVertices.emplace_back(_processVertices(i, vertices));
+    processVertices.emplace_back(futures::makeFutureWith(
```
Comment (romanatarango) [inline arangod/Pregel/Worker/Worker.cpp:370]: `processVertices` are rather `processedVertices`.

**Review by romanatarango — APPROVED** [review]
lgtm

File: arangod/Pregel/Worker/Worker.cpp (hunk 1 above)
Comment (jvolmer) [reply arangod/Pregel/Worker/Worker.cpp:790]: Thanks for the tipp, makeFuture is much better here :)

We decided that in Pregel we want to get rid of as many exceptions as possible and use ResultT instead. This makes it more obvious where errors could occur and force you to handle them directly.

**Pull request merged**